### PR TITLE
Removed pure_count and added property methods to _SinonStubCondition

### PR DESCRIPTION
**Pull request by jonathan-benn**
Fixes note35/sinon#40

### CODE REVIEW

**Comment by note35** [comment]
Overall it's a good change, and I appreciate that the code get benefit from your design obviously. (Clean code is always a good thing!)

For reducing the test code size, I believe call function **one** time is enough.

Besides, I recommend to check:

- test call count before call.

```python
@sinontest
def test_example1(self):
    stub = SinonStub(os, "system").onSecondCall().returns(42)
    self.assertEqual(pure_stub2.callCount, 0)
    os.system("cd")
    self.assertEqual(pure_stub2.callCount, 1)
```

- declare 2 pure spy/stub, call them separately, and test their call count.

```python
@sinontest
def test_example2(self)
    pure_stub1 = SinonStub().onSecondCall().returns(42)
    pure_stub2 = SinonStub().onSecondCall().returns(42)
    pure_stub1()
    pure_stub2()
    pure_stub2()
    self.assertEqual(pure_stub1.callCount, 1)
    self.assertEqual(pure_stub2.callCount, 2)
```

**Comment by jonathan-benn** [comment]
Hi Kir,

It's true that there was a lot of overlap in my newly created tests. I've tried to follow your suggestion and combine everything into one test.

--Jonathan

File: sinon/test/TestSinonStub.py [hunk 1]
```diff
@@ -596,3 +596,29 @@ def test443_withArgs_onCall_throws_does_not_reset_call_count(self):
...
+        self.assertEqual(os.system("cd"), None)
+        with self.assertRaisesRegexp(BaseException, 'Hello World'):
+            sinon.g.C_func()
+        self.assertEqual(sinon.g.C_func(), 42)
+        self.assertEqual(sinon.g.C_func(), None)
+        self.assertEqual(pure_stub.callCount, 3)
+        self.assertEqual(module_function_stub.callCount, 3)
+        self.assertEqual(function_stub.callCount, 3)
+        module_function_stub.restore()
+        function_stub.restore()
```
Comment (note35) [inline sinon/test/TestSinonStub.py]: Here you have no need to restore, the decorator `@sinontest` has finished it.

**Comment by note35** [comment]
Hi, merging into one test is good and separating into 3 (class/function/pure) could also be a good idea.

**Comment by jonathan-benn** [comment]
I have split one test into three tests

**Review by note35 — APPROVED** [review]
Thanks! LGTM!

**Pull request merged**